modules/gathering-bootstrap-diagnostic-data.adoc

```diff
@@ -0,0 +1,62 @@
+//
+// *support/troubleshooting-installations.adoc
+
+[id="gathering-bootstrap-diagnostic-data_{context}"]
+= Gathering bootstrap node diagnostic data
+
+Gather `bootkube.service` journald unit logs and container logs from the bootstrap node, when experiencing bootstrap related issues.
+
+.Prerequisites
+
+* The `ssh-agent` process is active on your computer, and you provided both the `ssh-agent` process and the installation program the same SSH key.
+* If you tried to install a cluster on infrastructure that you provisioned, you must have the fully-qualified domain names of the bootstrap node.
+
+.Procedure
+
+. If you have access to the bootstrap node's console, monitor the console until the node reaches the login prompt.
+
+. Verify ignition file configuration:
+.. Verify the bootstrap node ignition file URL:
++
+----
+curl -I http://<fully_qualified_domain_name>:<port>/bootstrap.ign  <1>
+----
+<1> The `-I` option returns the header only. If the ignition file is available on the specified URL, the command returns `200 OK` status. If it is not available, the command returns `404 file not found`.
++
+.. Query HTTP server logs on the serving host, to verify that the ignition file was received by the bootstrap node. For example, if you are using an Apache web server to serve ignition files:
++
+----
+$ grep -is 'bootstrap.ign' /var/log/httpd/access_log
+----
++
+If the bootstrap ignition file is received, the associated `HTTP GET` log message will include a `200 OK` success status, indicating that the request succeeded.
++
+.. If the ignition file was not received, check that the ignition files exist and have the appropriate file and web server permissions, on the serving host directly.
+
+. Verify the availability of the bootstrap node's assigned storage device.
+
+. Verify that the bootstrap node has been assigned an IP address from the DHCP server.
+
+. Collect `bootkube.service` journald unit logs from the bootstrap node. Replace `<cluster_name>.<sub_domain>.<domain>` with appropriate values:
++
+----
+$ ssh core@bootstrap.<cluster_name>.<sub_domain>.<domain> journalctl -b -f -u bootkube.service
+----
++
+[NOTE]
+====
+The `bootkube.service` log on the bootstrap node will output etcd `connection refused` errors, indicating that the bootstrap server is unable to connect to etcd on master nodes. Once etcd has started on each master node and the nodes have joined the cluster, the errors should stop.
+====
+
+. Collect logs from bootsrap node containers:
+.. Collect the logs using `podman` on the bootstrap node. Replace `<cluster_name>.<sub_domain>.<domain>` with appropriate values:
++
+----
+$ ssh core@bootstrap.<cluster_name>.<sub_domain>.<domain> 'for pod in $(sudo podman ps -a -q); do sudo podman logs $pod; done'
+----
+
+. If the bootstrap process fails, verify the following:
++
+* You can resolve `api.<cluster_name>.<base_domain>` from the installation host.
+* The load balancer proxies port 6443 connections to bootstrap and master nodes.
```

Comment:
Thanks Eric. That module is part of the same assembly (`support/troubleshooting-installations.adoc`) and is placed towards the start of the assembly. As such, customers should hopefully have covered that before getting to this module on gathering bootstrap data. I would like to be able to link to the module for reference within this module, unfortunately we are not able to include xrefs within modules at this point.

Also, because this module may be reused in the future in other assemblies, I can't include something like 'refer to the preceding section', as this won't make sense in other assemblies unless both modules are included and in the same order.

In light of all of that, I have added to line 62 in the module as follows:

~~~
* The load balancer proxies port 6443 connections to bootstrap and master nodes. Ensure that the proxy configuration meets {product-title} installation requirements.
~~~